DataRepo/tests/test_infusate_name_parser.py

```
@@ -147,7 +146,7 @@ def test_malformed_tracer_parsing_2(self):
 
     def test_malformed_tracer_parsing_3(self):
         # Test bad isotope pattern not silently skipped
-        name = "lys1,2,3-13C3,badlabel,19O2"
+        name = "1,2,3-13C3,badlabel,19O2"
         with self.assertRaisesRegex(TracerParsingError, "cannot be parsed"):
             _ = parse_infusate_name(name)
```

B. This is calling the wrong parser.  It should be calling `parse_isotope_string`, but instead, it is passing this isotope string to the `parse_infusate_name` method, which of course raises an exception.  I tested this string with the `parse_isotope_string` method in the shell and it does not generate an error as it should.  It silently skips malformed isotope strings:
```
In [1]: from DataRepo.utils import infusate_name_parser as inp

...

In [10]: inp.parse_isotope_string("1,2,3-13C3,badlabel,19O2")
Out[10]: 
[{'labeled_element': '13C',
  'labeled_count': 3,
  'labeled_positions': [1, 2, 3]},
 {'labeled_element': '19O', 'labeled_count': 2, 'labeled_positions': None}]
```

---

I will put in a regex of "disallowed characters" to catch this particular string, but it is not clear to me if the encoding couldn't just be malformed in another manner, which would have to rely on validation code.  If you or Lance has a replacement pattern for `ISOTOPE_ENCODING_PATTERN`, we can try that.

---

This can be solved by adding an anchor (e.g. perl's `\G`).  Not sure what the equivalent python re method is, but my suggestion on how to solve it was [here](https://github.com/Princeton-LSI-ResearchComputing/tracebase/pull/441#discussion_r883766290).

---

And this was one reason that I was opposed to using comma for multiple delimited items.  I just googled a bit and found that there's no viable way to do a perlre `\G` equivalent.  Work-arounds involve setting the `pos` (though I didn't dig into it) and have caveats.  It seems the most reliable way to solve would be by splitting on comma and conditionally rebuilding each expression.

---

Well, I suppose you could concatenate all the matches back together, and if is does not match the input, raise an error?  I would prefer if we don't delay moving forward, given the number of tracer example we have been given so far, assuming that curator are screening all inputs, and assuming that validation code is going to be layered on top of this.

---

That would work too. Less math.

---

tried it